### PR TITLE
gevent: bump to >=1.0

### PR DESCRIPTION
**Pull request by zhouyuan**
There are some DNS resolving issue on 0.13.6.
Bumping to 1.0 fixes the issues for me.

Signed-off-by: Yuan Zhou <yuan.zhou@intel.com>

### CODE REVIEW

**Comment by zhouyuan** [comment]
Hi @theanalyst here's the PR for issue #78

**Comment by theanalyst** [comment]
@zhouyuan Thank you very much! I'll give this a try to see if it solves the DNS issues we're seeing in teuthology-openstack

File: s3tests/readwrite.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def main():
             warmup_pool = gevent.pool.Pool(size=100)
             for file_name in file_names:
                 fp = next(files)
-                warmup_pool.spawn_link_exception(
+                warmup_pool.spawn(
```
Comment (theanalyst) [inline s3tests/readwrite.py]: I know `spawn_link_exception` is deprecated from 0.13, but is it supposed to be replaced by spawn? (iirc there is something like `link_exception`)

Comment (zhouyuan) [inline s3tests/readwrite.py]: @theanalyst 
Yes there need to have some exception handling code - this patch was just a quick trying

Comment (theanalyst) [inline s3tests/readwrite.py]: @zhouyuan ah ok thanks for taking a look

**Comment by dachary** [comment]
@oritwas should this be against ceph-master or master ? I'm not sure what the difference is ...

**Comment by dachary** [comment]
@zhouyuan it's a great patch as it will unblock running rgw tests on OpenStack :-) Would you mind adding Fixes: #13885 http://tracker.ceph.com/issues/13885 to the commit message for cross references purposes ?

**Comment by dachary** [comment]
@zhouyuan oritwas confirmed that master is the target, sorry for the noise

**Comment by zhouyuan** [comment]
@dachary no problem. I'm still trying to figure out how to add the exception handling code. Did you happen to run into any issue on 'greenlet exception' there or the current code is working fine?

**Comment by dachary** [comment]
@zhouyuan we did not run into problem. Given that it took about six months to figure this problem out, I suspect that if we leave room for uncertainty in this patch, it will be extremely hard to figure out when and if it manifests itself ;-)

Are there tests we can run that are guaranteed to cover all lines modified by this patch ? 

Note that I'm 100% ignorant of s3test and rgw, that does not help. @theanalyst and @oritwas may have a more informed opinion.

**Comment by robbat2** [comment]
+1 on merging from me; still would be better when the exception handling is added.

**Comment by dachary** [comment]
@zhouyuan to summarize http://tracker.ceph.com/issues/13885#note-11 : once this is merged into master, it should be merged into ceph-master and it will be used by http://pulpito.ceph.com/?suite=rgw&branch=master

Do you think this pull request is ok to be merged or do you need more time to work on it ?

**Comment by zhouyuan** [comment]
@dachary thanks for the notes! 
I think we could merge this first. @oritwas and @robbat2 are both OK for this.

**Comment by theanalyst** [comment]
Running ```teuthology-suite --suite rgw --suite-branch gevent1 --machine-type plana,burnupi,mira --email abhishek@suse.com --ceph master --filter s3test --subset 0/200 ```

http://pulpito.ceph.com/abhi-2016-04-12_06:24:25-rgw-master---basic-multi/

**Comment by theanalyst** [comment]
Both the suites were dead, rescheduled them
> 
> Running teuthology-suite --suite rgw --suite-branch gevent1 --machine-type plana,burnupi,mira --email abhishek@suse.com --ceph master --filter s3test --subset 0/200
> 
> http://pulpito.ceph.com/abhi-2016-04-12_06:24:25-rgw-master---basic-multi/

http://pulpito.ceph.com/abhi-2016-05-02_01:49:58-rgw-master---basic-multi/
> And teuthology-suite --suite rgw --machine-type plana,burnupi,mira --email abhishek@suse.com --ceph master --filter s3test --subset 0/200
> 
> http://pulpito.ceph.com/abhi-2016-04-12_06:26:07-rgw-master---basic-multi/
> 
http://pulpito.ceph.com/abhi-2016-05-02_01:47:11-rgw-master---basic-multi/

**Comment by theanalyst** [comment]
Both the suites were dead, rescheduled them again
> 
> Running teuthology-suite --suite rgw --suite-branch gevent1 --machine-type plana,burnupi,mira --email abhishek@suse.com --ceph master --filter s3test --subset 0/200
>> 
>> http://pulpito.ceph.com/abhi-2016-04-12_06:24:25-rgw-master---basic-multi/

> http://pulpito.ceph.com/abhi-2016-05-02_01:49:58-rgw-master---basic-multi/
http://pulpito.ceph.com/abhi-2016-05-18_03:46:49-rgw-master---basic-multi/
> And teuthology-suite --suite rgw --machine-type plana,burnupi,mira --email abhishek@suse.com --ceph master --filter s3test --subset 0/200
>> 
>> http://pulpito.ceph.com/abhi-2016-04-12_06:26:07-rgw-master---basic-multi/
http://pulpito.ceph.com/abhi-2016-05-18_03:45:22-rgw-master---basic-multi/
>http://pulpito.ceph.com/abhi-2016-05-02_01:47:11-rgw-master---basic-multi/

**Comment by zhouyuan** [comment]
@theanalyst It looks like these tests died due to timeout when scheduling? the logs are all missing(404) there

**Comment by theanalyst** [comment]
@zhouyuan yeah so far none of the tests have actually failed (or passed)as they have never been able to run at all, hoping this run might succeed :)

**Comment by theanalyst** [comment]
Rescheduled  again: 
gevent: http://pulpito.ceph.com/abhi-2016-06-01_02:49:26-rgw-master---basic-multi/
normal:  http://pulpito.ceph.com/abhi-2016-06-01_02:49:03-rgw-master---basic-multi/

**Comment by theanalyst** [comment]
Rescheduled  again: with higher priority
gevent: http://pulpito.ceph.com/abhi-2016-07-14_01:52:38-rgw-master---basic-multi/
normal: http://pulpito.ceph.com/abhi-2016-07-14_01:53:06-rgw-master---basic-multi/

**Comment by theanalyst** [comment]
rescheduled only on smithi:
http://pulpito.ceph.com/abhi-2016-08-02_02:05:07-rgw-master---basic-smithi/
http://pulpito.ceph.com/abhi-2016-08-02_02:07:15-rgw-master---basic-smithi/

**Pull request merged**